### DISCUSSION TITLE
How to run the test from the Cookbook ?

### DISCUSSION THREAD

**Discussion by jlecordier** [PHP-CS-Fixer/PHP-CS-Fixer#8032] [Q&A] [answered]
Hi,

The cookbook states that one should run the test by using `phpunit tests/Fixer/Comment/RemoveCommentsFixerTest.php`.
However, out of the box, this command does not work (`bash: phpunit: command not found`).
- Should having phpunit installed globally be an explicit prerequisite ?
- Should `composer exec phpunit ...` be used ?
- Should `./vendor/bin/phpunit ...` be used ?
- Should `composer run test ...` be used ?
- Should something else be used ?

Thanks !

**Comment by Wirone** [comment]
It's hard to tell how PHPUnit is installed in a project where custom fixer is developed. That's why it's a generic `phpunit` command, which makes it clear what is the usage, but it needs to be aligned with actual project structure.

**Reply by jlecordier** [reply]
The cookbook takes place inside the PHP-CS-Fixer repo, should we not follow how "official" fixers are developed ?
And what if I want to develop an "official" fixer ? Is the procedure written somewhere else than the cookbook ?

**Reply by Wirone** [reply, marked as answer]
You're right, the cookbook is focused on developing new rules within the Fixer repo (fork), so it should follow the real commands. Maybe it's a good idea to state that after implementing a new rule (with tests) whole `composer qa` must pass, not only tests 🙂. But for PHPUnit it should be `vendor/bin/phpunit`.
